lib/KBox/src/pages/NavigationPage.cpp

```diff
@@ -0,0 +1,305 @@
+/*
+  The MIT License
+
+  Copyright (c) 2016 Thomas Sarlandie thomas@sarlandie.net
+
+  Permission is hereby granted, free of charge, to any person obtaining a copy
+  of this software and associated documentation files (the "Software"), to deal
+  in the Software without restriction, including without limitation the rights
+  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+  copies of the Software, and to permit persons to whom the Software is
+  furnished to do so, subject to the following conditions:
+
+  The above copyright notice and this permission notice shall be included in
+  all copies or substantial portions of the Software.
+
+  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+  THE SOFTWARE.
+*/
+
+/* AutoPilot functionality can exist at several levels.  In this initial iteration, the design 
+ focused on heading the boat on a defined course that is automatically set when AP functionality 
+ is turned on (referred to as "Heading" mode below.  Later on we could introduce an additional 
+ mode when a course to a waypoint is provided by a NMEA-connected device.  In addition, we have 
+ "Dodge" mode which suspends rudder actions temporarily in order to manually steer the boat around
+ an object or to manually put it back on course.  Waypoint functionality will be added at a later time. RIGM
+ */
+
+#include "NavigationPage.h"
+#include "MFD.h"
+#include <stdio.h>
+#include "KBoxDebug.h"
+
+
+bool apMode = false;
```

Comment:
This is where my knowledge of c++ falls short.  How would I make these variables a member of the class?

With respect to having them as an enum, that would present some challenges.  I set them up as separate booleans because because they are not all mutually exclusive.  If we are in heading mode and switch to dodge mode, we need to know to go back to heading mode when we come out of dodge mode, rather than Waypoint mode.  Heading and Waypoint are mutually exclusive, but Dodge is not.  For now I suggest we stick with the booleans.